source/common/http/path_utility.cc

```diff
@@ -85,5 +85,63 @@ absl::string_view PathUtil::removeQueryAndFragment(const absl::string_view path)
   return ret;
 }
 
+absl::optional<std::string> PathTransformer::mergeSlashes(absl::string_view original_path) {
+  const absl::string_view::size_type query_start = original_path.find('?');
+  const absl::string_view path = original_path.substr(0, query_start);
+  const absl::string_view query = absl::ClippedSubstr(original_path, query_start);
+  if (path.find("//") == absl::string_view::npos) {
+    return original_path.data();
```

Comment:
there probably isn't that much of a concern of returning a pointer at this point in the code since NULs should be gone, but consider explicitly constructing with `original_path.size()`?